### PR TITLE
[#162263] Multi-Facility - 5c When an admin places a cross-core order, update valid payment source list

### PR DESCRIPTION
**Pull request by LeticiaErrandonea**
# Release Notes
* When an admin places a cross-core order, update valid payment source list.

# Screenshot
image

image

# Additional Context

# Accessibility
- [x] Did you scan for accessibility issues?
- [x] Did you check our accessibility goal checklist?
- [ ] Did you add accessibility [specs](https://github.com/dequelabs/axe-core-gems/blob/develop/packages/axe-core-rspec/README.md)?

### CODE REVIEW

File: app/lib/ability.rb [hunk 1]
```diff
@@ -168,6 +168,10 @@ def facility_director_abilities(user, resource, controller)
   def facility_administrator_abilities(user, resource, controller)
     can :available_for_cross_core_ordering, Product if controller.is_a?(ProductsController)
 
+    if controller.is_a?(FacilityAccountsController) && user.facility_administrator_of_any_facility?
+      can [:accounts_available_for_order, :show], Account
```
Comment (LeticiaErrandonea) [inline app/lib/ability.rb:172]: Had to add `show` as it's required by the `authorize_resource`

File: spec/system/admin/adding_to_an_order_spec.rb [hunk 2]
```diff
@@ -287,9 +288,12 @@
         expect(page).not_to have_content("Cross Core Project ID")
         expect(page.has_selector?("option", text: cross_core_product_facility.name, visible: false)).to be(true)
         expect(page.has_selector?("option", text: product.name, visible: false)).to be(true)
+        expect(page.has_selector?("option", text: other_account.to_s, visible: false)).to be(false)
 
         select_from_chosen facility2.name, from: "add_to_order_form[facility_id]"
         select_from_chosen cross_core_product_facility2.name, from: "add_to_order_form[product_id]"
+        # This fails because the account is not valid for the facility
+        # select_from_chosen other_account.to_s, from: "Payment Source"
```
Comment (LeticiaErrandonea) [inline spec/system/admin/adding_to_an_order_spec.rb:296]: For some reason `other_account` is not available for the facility, looks like a setup issue

Comment (LeticiaErrandonea) [reply spec/system/admin/adding_to_an_order_spec.rb]: Fixed!

File: app/controllers/facility_accounts_controller.rb [hunk 3]
```diff
@@ -158,4 +167,14 @@ def build_account
     ).build
   end
 
+  def set_facility_accounts_for_user
+    order_id = params[:order_id]
+    order = Order.find(order_id) if order_id
+
+    return unless order
```
Comment (giladshanan) [inline app/controllers/facility_accounts_controller.rb:174]: What is the UX in this case?  should we return an empty collection instead?

File: app/views/facility_orders/_merge_order_form.html.haml [hunk 4]
```diff
@@ -31,11 +32,19 @@
...
-          collection: @add_to_order_form.available_accounts,
-          include_blank: !@order.account.active?,
-          hint: @order.account.active? ? "" : t(".original_account_inactive", account: @order.account),
-          input_html: { class: "js--chosen" }
+        - if enable_cross_core_changes
+          = f.input :account_id,
+            collection: @add_to_order_form.available_accounts,
+            hint: @order.account.active? ? "" : t(".original_account_inactive", account: @order.account),
+            include_blank: !@order.account.active?,
+            input_html: { class: "js--edit-order__account js--chosen",
```
Comment (giladshanan) [inline app/views/facility_orders/_merge_order_form.html.haml:40]: If this line is the only difference between the two cases, maybe we can set that value separately like:
```
input_html = enable_cross_core_changes ? { class: "js--edit-order__account js--chosen", data: { "include-blank" => !@order.account.active? } } : { class: "js--chosen" }
```

File: spec/system/admin/adding_to_an_order_spec.rb [hunk 5]
```diff
@@ -275,6 +275,7 @@
 
     context "with admin role" do
       let(:user) { create(:user, :facility_administrator, facility:) }
+      let!(:other_account) { create(:account, :with_account_owner, type: "CreditCardAccount", owner: order.user, description: "Other Account", facility: facility2) }
```
Comment (giladshanan) [inline spec/system/admin/adding_to_an_order_spec.rb:278]: how about:
```suggestion
      let!(:facility2_account) { create(:account, :with_account_owner, type: "CreditCardAccount", owner: order.user, description: "Other Account", facility: facility2) }
```

**Review by giladshanan — APPROVED** [review]
A few small suggestions, nothing blocking

File: app/controllers/facility_accounts_controller.rb (hunk 3 above)
Comment (LeticiaErrandonea) [reply app/controllers/facility_accounts_controller.rb]: UX was an ugly error in the console 😅 

Fixed now, thanks!

**Pull request merged**